<?xml version="1.0"?>
<launch>
  <!-- Profile: ultra (lowest latency) | stable (more tolerant) -->
  <arg name="profile" default="ultra" />
  <arg name="buffer_mode" default="none" /> <!-- none|auto -->
  <arg name="drop_on_late_src" default="false" />
  <arg name="jitter_drop_on_late" default="false" />
  <arg name="jitter_latency_ms" default="" /> <!-- empty = use latency_ms per camera -->
  <arg name="parse_alignment" default="true" />
  <arg name="parse_config_interval" default="1" />
  <arg name="publish_rate_hz" default="120.0" />
  <arg name="use_watchdog" default="false" />
  <arg name="stale_timeout_sec" default="2.0" />

  <!-- Cameras: edit IPs and password or override via rosparam yaml -->
  <node pkg="ip_camera"
        type="gst_rtsp_ros_streamer.py"
        name="gst_rtsp_ros_streamer"
        output="screen"
        respawn="true"
        respawn_delay="3">
    <param name="profile" value="$(arg profile)" />
    <param name="buffer_mode" value="$(arg buffer_mode)" />
    <param name="drop_on_late_src" value="$(arg drop_on_late_src)" />
    <param name="jitter_drop_on_late" value="$(arg jitter_drop_on_late)" />
    <param name="jitter_latency_ms" value="$(arg jitter_latency_ms)" />
    <param name="parse_alignment" value="$(arg parse_alignment)" />
    <param name="parse_config_interval" value="$(arg parse_config_interval)" />
    <param name="publish_rate_hz" value="$(arg publish_rate_hz)" />
    <param name="use_watchdog" value="$(arg use_watchdog)" />
    <param name="stale_timeout_sec" value="$(arg stale_timeout_sec)" />

    <rosparam param="cameras">
      - name: camera_1
        ip: 192.168.0.171
        username: admin
        password: zjsxmfhf
        stream: stream2
        topic: /camera/camera_1/image_raw
        frame_id: camera_1_link
        transport: udp
        latency_ms: 80
        decoder: auto
      - name: camera_2
        ip: 192.168.0.195
        username: admin
        password: zjsxmfhf
        stream: stream2
        topic: /camera/camera_2/image_raw
        frame_id: camera_2_link
        transport: udp
        latency_ms: 80
        decoder: auto
    </rosparam>
  </node>
</launch>


